05-tasks-and-events.md

```diff
@@ -34,85 +32,140 @@ Tasks MUST follow those rules:
 
-This RFC lists available tasks that MUST be supported by syncers, unless noted otherwise, and their respective outputs: blockchain events. A task produces zero or more `events` during its lifetime.
+This RFC lists available tasks that MUST be supported by syncers, unless noted otherwise, and their respective output messages: `blockchain events`. A task produces zero or more `events` during its lifetime.
 
 Every task is accompanied with an `id`, a positive integer which fits into the 32-bit signed integer space. This integer has no rules on its generation or any pattern required by its usage. Every emitted event has its own `id` field corresponding to the task which caused it.
 
 The `error` event is valid for every single task, and it takes the task's ID and provides an integer `code`, as well as optionally a string `message`.
 
 This document frequently references epochs, whose definition is dependent on the protocol in question. For Bitcoin, Monero, and other blockchain-based systems, the epoch is the block height. For systems without the context of a block height, Unix timestamps SHOULD be used.
 
+### The `abort_id` Task
+Task consists in aborting the task with id `id`.
+
+Data: 
+- `id`: the task `id` that shall be aborted
+
+To update a task with id `id`, Syncer must receive an `abort_id` task aborting with `id` parameter, and subsequently submit a new instance of this task which shall have a new `id`. 
+
 ### The `watch_height` Task
 
-`watch_height` asks the syncer for notifications about updates to the blockchain's height. This task MUST be implemented for any coin with a blockchain. This task MAY be implemented for any coin without a blockchain. If it is not implemented, an error event must be sent in response to any attempt to start this task.
+`watch_height` asks the syncer for notifications about updates to the blockchain's highest block. This task MUST be implemented for any coin with a blockchain. This task MAY be implemented for any coin without a blockchain. If it is not implemented, an error event must be sent in response to any attempt to start this task.
 
 Required parameters are:
-* `lifetime`: Epoch at which the syncer SHOULD drop this task. Until then, barring another instance of this task, this task MUST be maintained. When another instance appears, syncers MAY only keep the most recent one.
+* `lifetime`: Epoch at which the syncer SHOULD drop this task. Until then, this task MUST be maintained, barring the case where an `abort_id` task aborts it. 
 
 Parameters may be added to specify which blockchain, in order to support any network utilizing multiple.
 
-When the height changes, a `height_changed` event is emitted. It contains:
-* `height`: Current blockchain height.
-
-Upon task reception, syncers MUST send a `height_changed` event immediately to signify the current height.
```

Comment:
This shouldn't be removed, as otherwise there is no way to know "instantly" what is the current height.

Reply:
it now is on the events section below